pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -156,37 +148,46 @@ func (nc *NodeController) generateCIDRs(num int) util.StringSet {
 	return res
 }
 
-// For each node from newNodes, finds its current spec in registeredNodes.
-// If it is not there, it gets a new valid CIDR assigned.
-func (nc *NodeController) reconcilePodCIDRs(newNodes, registeredNodes *api.NodeList) *api.NodeList {
-	registeredCIDRs := make(map[string]string)
-	availableCIDRs := nc.generateCIDRs(len(newNodes.Items) + len(registeredNodes.Items))
-	for _, node := range registeredNodes.Items {
-		registeredCIDRs[node.Name] = node.Spec.PodCIDR
-		availableCIDRs.Delete(node.Spec.PodCIDR)
+// reconcilePodCIDRs looks at each node and assigns it a valid CIDR
+// if it doesn't currently have one.
+func (nc *NodeController) reconcilePodCIDRs(nodes *api.NodeList) {
+	glog.V(4).Infof("Reconciling pods cidrs for %d nodes", len(nodes.Items))
+	// TODO(roberthbailey): This seems inefficient. Why re-calculate CIDRs
+	// on each sync period?
+	availableCIDRs := generateCIDRs(nc.clusterCIDR, len(nodes.Items))
+	for _, node := range nodes.Items {
+		if node.Spec.PodCIDR != "" {
+			glog.V(4).Infof("CIDR %s is already being used by node %s", node.Spec.PodCIDR, node.Name)
+			availableCIDRs.Delete(node.Spec.PodCIDR)
+		}
 	}
-	for i, node := range newNodes.Items {
-		podCIDR, registered := registeredCIDRs[node.Name]
-		if !registered {
-			podCIDR, _ = availableCIDRs.PopAny()
+	for _, node := range nodes.Items {
+		if node.Spec.PodCIDR == "" {
+			podCIDR, found := availableCIDRs.PopAny()
```

Comment:
Yes, that's why we only persist it if nc.configureNodeCIDR returns success (b/c that's what configures the route on GCE).